### PR TITLE
Add limited FSDP correctness to torchdynamo benchmark

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89457
* #89520
* #89523
* #89330
* __->__ #89469
* #89515

- Does not do recursive wrapping
- Only supports accuracy bench
- Mainly useful for sweeping over models for correctness, in part
  to evaluate whether dynamo support for FSDP is breaking anywhere

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Review by davidberard98 — APPROVED** [review]
LGTM!

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -1091,6 +1092,10 @@ def deepcopy_and_maybe_ddp(model):
             model = copy.deepcopy(model)
             if self.args.ddp:
                 model = DDP(model, find_unused_parameters=True)
+            elif self.args.fsdp:
+                model = FSDP(model)
```
Comment (aazzolini) [inline benchmarks/dynamo/common.py:1096]: Just want to understand here -- this is a single wrap, correct?

Comment (aazzolini) [reply benchmarks/dynamo/common.py:1096]: Did we decide that we're not doing any custom wrapping?

**Review by aazzolini — APPROVED** [review]

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (wconstab) [reply benchmarks/dynamo/common.py:1096]: no, we still are. this test is used for sweeping all of the benchmark suite models.  for that purpose it’s not feasible to configure custom wrapping, but the sweep helps find bugs with dynamo handling of those models inside FSDP wrapper. The main case i expect to catch is that UnspecializedNNModule misses some feature used by a model.

**Pull request closed without merging**